### DISCUSSION TITLE
Allow using reusable workflows in private repositories for organizations

### DISCUSSION THREAD

**Discussion by vitalii-dubovoi** [community/community#27872] [Actions] [answered]
Hello,

As I understand there is a limitation for re-using workflow from another repository, it must be public. Is it possible to add support for Team plans to use workflows from the private repo. In this case, we can have one centralized repo that stores all needed workflows pipelines

For now, it only works for Enterprise plans.

**Comment by Yohnah** [comment]
I think it could be a good feature to perform complex pipelines without having duplicated codes (workflow codes)

**Reply by dgokcin** [reply]
any updates in this issue?

**Reply by aavileli** [reply]
Why lock it to github enterprise  ?

**Reply by vitalii-dubovoi** [reply]
I found a workaround with this action - https://github.com/BetaHuhn/repo-file-sync-action
It allows to sync of files from one repo to others

**Reply by dgokcin** [reply]
@vitalii-dubovoy just did a quick POC with the action you suggested. This is going to be very helpful since I do not have to do a update in 30 repositories whenever I do a small enhancement or find a new action. I do not know what GitHub has implemented on their enterprise version to this pain but I believe it works really well with auto labels and PRs. Thank you very much. 🎉

**Reply by Breich90** [reply]
The documentation here is incredibly confusing, but from what I've gathered only `internal` type repositories can house reusable workflows.  My company pays for Enterprise however we do not have an Enterprise account made yet (we're single organization at the moment).  So, we can only create `public` or `private` repositories... and thus, reusable workflows are not found despite pointing to a `private` repo on an Enterprise billed account.  The workaround action is probably the answer here unless you're able to create `internal` repositories.

**Comment by fishpen0** [comment, marked as answer]
This looks to be on the roadmap but keeps getting bumped: https://github.com/github/roadmap/issues/51
